syncd/ConcurrentQueue.h

```diff
@@ -0,0 +1,127 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
```

Comment:
@lguohan do we allow each file to have different license?

Reply:
no

Reply:
Are you saying none of the files that we are trying to upstream can have license? 

cc: @mikeberesford @bhagatgit

Reply:
I think we have a license for entire repo not per single file

Reply:
@lguohan  @kcudnik  Can you provide more details on the licensing. 
On repos that have multiple contributors, does this mean that the contributor that started the repo would add license and others would not ?

Reply:
@kcudnik while license issue is being resolved, can you please provide other review comments that you may have so that I can work on those in parallel for this PR and other files that will come in next few PRs? Thank you.

Reply:
sure, please also fix build errors

Reply:
Thanks. Addressed the review comments. Build issue is because of spell checker (aspellcheck.pl) failing due to it complaining about LLC in "Google LLC" in the copyright header. Not sure why LLC is causing spell check issue though.

Reply:
@kcudnik We are still getting the licensing header issue resolved internally. Given that it is taking time, I have started generating the remaining commits to this PR. Can you please review them too? I dont know if I was supposed to generated separate PR for each commit or all the commits can be generated in same PR. Given these commits are dependent, generating multiple commits in same PRs sounds reasonable. Thanks.

Reply:
Think you will hav3 still multiple classes and many many code in one pr,  I would advise to have separate pr for specific modules

Reply:
Just to confirm, what do you mean exactly by separate PR here? Does it mean each commit should have its own PR, for example: commit 1 in PR https://github.com/sonic-net/sonic-sairedis/pull/XXXX, commit 2 in https://github.com/sonic-net/sonic-sairedis/pull/YYYY, commit 3 in https://github.com/sonic-net/sonic-sairedis/pull/ZZZZ, and so on?

Or you are saying multiple commits in this same PR is ok but each commit should not have multiple classes?

A lot of code depend on each other. For example: commit 2 in this PR depends on commit 1 in the PR. So, if we generate separate separate PR for each commits, they will not compile since dependency will not be present.

Reply:
no, not the commit, but if you are planning to add many big classes, it make sense to add each class separately, since we squash later all the commits to 1 commit on master, and if you add 40 files in one commit that will be overkill and unmanagable